src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterKeyValuePair.cs

```diff
@@ -41,12 +45,14 @@ internal sealed class JsonKeyValuePairConverter<TKey, TValue> : JsonConverter<Ke
             string propertyName = reader.GetString()!;
             if (propertyName == KeyName)
             {
-                k = ReadProperty<TKey>(ref reader, typeToConvert, options);
+                reader.Read();
```

Comment:
Yes since this is a value converter and thus a single block for the reader (isFinalBlock) -- a false here shouldn't happen.

Reply:
"value" is an overload term. With your explanation, it makes perfect sense that `isFinalBlock = true` for these cases :)

It may seem like I am overly nitpicking (and I very well might be), but given one of the benefits of this PR is code simplicity and making it easier to maintain/evolve, it would be great if all this contextual knowledge (about state info elsewhere in the code influencing decisions in different parts) was captured in comments, `Debug.Asserts` with messages, etc. Particularly the **"why?"**

I bet weeks/months from now, any reader of the code would have similar **"whys?"** and have to re-infer that it makes sense to do so.

Up to you, ofc.

Reply:
I added ReadWithVerify() and applied to all Read() where we ignore the return value like this